src/imperative/imperative_utils.cc

```diff
@@ -186,7 +186,7 @@ void NaiveRunGraph(
       Imperative::Get()->InvokeOp(ctx, node.source->attrs, ndinputs, ndoutputs,
                                   req, dispatch_mode, state);
       for (size_t j = 0; j < ndoutputs.size(); ++j) {
-        if (ndoutputs[j]->shape().ndim() == 0) {
+        if (ndoutputs[j]->shape().ndim() == -1) {
```

Comment:
nit: Better to use `shape_is_known` for readability. More precisely, it should be `shape_is_known` instead of just `ndim_is_known`, right?

Reply:
Good idea!